ui/decredmaterial/editor.go

```diff
@@ -81,88 +91,128 @@ func (t *Theme) Editor(editor *widget.Editor, hint string) Editor {
 			},
 		},
 	}
+	e.TitleLabel.Text = m.Hint
+
+	return &e
+}
+
+func (e *Editor) calculateDims(gtx layout.Context) {
+	layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			e.titleLabelDims = e.TitleLabel.Layout(gtx)
+			return e.titleLabelDims
+		}),
+		layout.Rigid(func(gtx C) D {
+			e.editorDims = e.EditorStyle.Layout(gtx)
+			e.editorLineHeight = e.editorDims.Size.Y
+			return e.editorDims
+		}),
+		layout.Rigid(func(gtx C) D {
+			e.lineDims = layout.Inset{
+				Top:    unit.Dp(4),
+				Bottom: unit.Dp(4),
+			}.Layout(gtx, func(gtx C) D {
+				return e.editorLine(gtx)
+			})
+			return e.lineDims
+		}),
+	)
 }
 
-func (e Editor) Layout(gtx layout.Context) layout.Dimensions {
+func (e *Editor) calculateEditorHeight(gtx layout.Context) {
+	r := strings.NewReader(e.Editor.Text())
+	lines, _ := e.shaper.Layout(e.Font, fixed.I(gtx.Px(e.TextSize)), gtx.Constraints.Max.X, r)
+	e.editorDims.Size.Y = e.editorLineHeight * len(lines)
+}
+
+// Layout renders the editor to screen. The editor line is able to retain
+// it's relative position whether or not the hint or title labels are displayed
+// or not because their dimensions are pre-calculated before hand
+func (e *Editor) Layout(gtx layout.Context) layout.Dimensions {
+	if !e.hasCalculated {
+		e.calculateDims(gtx)
+		e.hasCalculated = true
+	}
+
+	if !e.Editor.SingleLine {
+		e.calculateEditorHeight(gtx)
+	}
+
 	e.handleEvents()
 	if e.IsVisible {
 		e.flexWidth = 0.93
 	}
+
 	if e.Editor.Focused() || e.Editor.Len() != 0 {
-		e.TitleLabel.Text = e.Hint
-		e.LineColor = color.RGBA{41, 112, 255, 255}
 		e.Hint = ""
 	}
 
+	if !e.Editor.Focused() {
+		e.Hint = e.StartHint
+	}
+
```

Comment:
since the Editor method isn't supposed to return a pointer, changing the value of e.Hint when focused would only change it temporarily. This means there wouldn't be a need to keep track of the previous "Hint" with the "StartHint" field. The hint value wouldn't just go back to the hint passed as an argument to the editor when not focused.